### PR TITLE
Clarify that napari should be installed

### LINKED ISSUE #4099: Your First Plugin guide needs more guidance around napari requirements
## 📚 Documentation

If you follow the [Your First Plugin](https://napari.org/plugins/stable/first_plugin.html) guide faithfully, running `napari` at the end will typically fail. That's because of the following steps.

1. Recommend creating a fresh virtual environment for your plugin (e.g. using conda).
2. Implement a plugin that depends on `napari` (specifically `napari.utils.notifications`). The dependency should not be `napari[all]` to follow [the best practice around depending on specific Qt backends](https://napari.org/plugins/stable/best_practices.html#don-t-include-pyside2-or-pyqt5-in-your-plugin-s-dependencies).
3. Calling `pip install -e .` then running `napari`.

We discussed a couple of solutions in the plugin working group. The one we preferred is to include a step like `pip install napari[all]` in the [Before you start](https://napari.org/plugins/stable/first_plugin.html#before-you-start) section to ensure that napari is installed and be run from that environment.

### PR DESCRIPTION
**Pull request by andy-sweet**
# Description
This adds a small explanation to the *Your First Plugin* guide that you need a full napari installation in order to use and test your plugin. See the linked bug for more details.

## Type of change
- [ ] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# References
Closes #4099 

# How has this been tested?
- [x] I built the docs locally using `jb build docs`

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

**Comment by andy-sweet** [comment]
Feel free to suggest any rewording. I tried to be exact and explanatory, but this was a little longer than I hoped. I'm a little worried that it puts too much at the top of the guide, so maybe we should consider moving this down to the bottom?

Also, I only provided the `pip` instruction using a `sh` code block because the syntax should be the same for Windows `bat`. Let me know if I should add an extra tab to match some other commands.

File: docs/plugins/first_plugin.md [hunk 1]
```diff
@@ -21,6 +21,22 @@ recommended**). If you are new to virtual environments, we recommend
 [installing miniconda][miniconda] and [creating a new environment with
 python][python_env].
 
+Even though your plugin does not need to directly depend on the `napari`
+package, and
```
Comment (tlambert03) [inline docs/plugins/first_plugin.md]:
```suggestion
Even though plugins don't necessarily need to list `napari` as a direct dependency, and
```

File: docs/plugins/first_plugin.md [hunk 2]
```diff
@@ -21,6 +21,22 @@ recommended**). If you are new to virtual environments, we recommend
...
+[should not depend on a specific Qt backend](best-practices-no-qt-backend),
+you will need a full installation of napari in your active Python
```
Comment (tlambert03) [inline docs/plugins/first_plugin.md]:
```suggestion
you will need a working installation of napari in your active Python
```

**Review by tlambert03 — APPROVED** [review]
LGTM thanks!  tiny/optional wording suggestions

File: docs/plugins/first_plugin.md (hunk 1 above)
Comment (tlambert03) [reply docs/plugins/first_plugin.md]: (mostly suggesting the "necessarily", since we later go on to demonstrate a plugin that _does_ need to list napari)

**Comment by tlambert03** [comment]
> Also, I only provided the pip instruction using a sh code block because the syntax should be the same for Windows bat. Let me know if I should add an extra tab to match some other commands.

i never quite know when to just direct them to one single "how to install napari" link, vs including it everywhere.  though it is nice to mention the min version here (but anyone installing by the time they read this will get the min required version)

File: docs/plugins/first_plugin.md [hunk 3]
```diff
@@ -21,6 +21,22 @@ recommended**). If you are new to virtual environments, we recommend
...
+environment to use and test your plugin.
+If napari is not already installed and working, consider installing it
+with `pip`
+
+```sh
+pip install "napari[all]>=0.4.13"
+```
+
+where `[all]` means that the default Qt backend will be installed and
+`0.4.13` is the minimum version of napari that supports `npe2`.
```
Comment (DragaDoncila) [inline docs/plugins/first_plugin.md]:
```suggestion
`0.4.13` is the minimum version of napari that supports `npe2` If you have never installed napari before, you may wish to look at our more detailed [installation guide](link-to-guide).
```

I'm wary of providing such succinct instructions for installation to new users so I'd like to link out to the full guide here.

**Review by DragaDoncila — CHANGES_REQUESTED** [review]

File: docs/plugins/first_plugin.md (hunk 3 above)
Comment (tlambert03) [reply docs/plugins/first_plugin.md]: sounds good.  maybe let's just do that for the whole thing?  That is:

> Even though your plugin does not need to directly depend on the napari package, and [should not depend on a specific Qt backend](https://github.com/napari/napari/blob/e4cb8ccd60d6973a585191e053346dc5aad25bad/docs/plugins/best-practices-no-qt-backend), you will need a full installation of napari in your active Python environment to use and test your plugin.  See the [installation guide]() if this is your first time installing napari.

?

Comment (DragaDoncila) [reply docs/plugins/first_plugin.md]: Yup, I think that works!

Comment (andy-sweet) [reply docs/plugins/first_plugin.md]: Thanks for the feedback @DragaDoncila. I used the wording from @tlambert03, so I think this is resolved now.

**Review by DragaDoncila — APPROVED** [review]
Thanks @andy-sweet LGTM!

**Review by Carreau — APPROVED** [review]

**Comment by tlambert03** [comment]
thanks!

**Pull request merged**